### DISCUSSION TITLE
If I lose connectivity from bitcoind and later regain, will Eclair reconnect automatically?

### DISCUSSION THREAD

**Discussion by fiatjaf** [ACINQ/eclair#2015] [Q&A] [answered]
I run my bitcoind node at home and Eclair on a VPS, and sometimes my home computer shuts down for various reasons, or I lose connectivity for some minutes. After that I always restart Eclair on the VPS because it apparently isn't able to pick up the connection to bitcoind again after it comes back, but this is just an impression based on not much more than nothing.

Should I instead rely on Eclair getting back together on its own? That would be much better.

**Comment by t-bast** [comment, marked as answer]
> it apparently isn't able to pick up the connection to bitcoind again after it comes back

Can you detail what you mean exactly by that?
Which connection? ZMQ or RPC calls?

For ZMQ, take a look at logs from the `ZmqActor` in your file, you should see logs about connection/disconnection (https://github.com/ACINQ/eclair/blob/498e9a7db1deb5274380ee8dd47e78fef95a854f/eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/zmq/ZMQActor.scala#L84). The underlying ZMQ library should handle automatic reconnection, if it doesn't it's worth investigating.

As for RPC calls, those don't require a long-lived connection, so they should just work when your node is back up.

AFAIK you shouldn't need to restart eclair (and if that's not the case right now, we should improve it).

**Reply by fiatjaf** [reply]
My impression was just based on failed payments (due to my Eclair node being too late on the chain in relation to other nodes) happening multiple minutes after bitcoind connectivity was back, which made me think it might not have picked up the connection back and was maybe not syncing itself to the current chain state.

But it was just a very loose impression, I probably didn't give it enough time. I'll keep watching this and report if something is not great, but should be fine.

Thanks!
